Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -1220,25 +1222,41 @@ private boolean readAndStripParam(Uri data, Activity activity) {
             // Capture the intent URI and extra for analytics in case started by external intents such as  google app search
             try {
                 if (data != null) {
+                    boolean foundSchemeMatch = false;
+                    boolean foundPathMatch = false;
+
                     if (externalUriWhiteList_.size() > 0) {
-                        String uriString = data.toString();
-                        boolean foundWhiteListMatch = false;
+                        String uriScheme = data.getScheme();
+
                         for (String whiteListScheme : externalUriWhiteList_) {
-                            if (uriString.contains(whiteListScheme)) {
-                                foundWhiteListMatch = true;
+                            if (uriScheme.equals(whiteListScheme)) {
+                                foundSchemeMatch = true;
                                 break;
                             }
                         }
-                        if (foundWhiteListMatch) {
-                            sessionReferredLink_ = data.toString();
-                            prefHelper_.setExternalIntentUri(data.toString());
+                    } else {
+                        foundSchemeMatch = true;
+                    }
+
+                    if (externalPathWhiteList_.size() > 0) {
+                        String uriPath = data.getPath();
+
+                        for (String whiteListPath : externalPathWhiteList_) {
```

Comment:
can we simplify this for loop with
`externalUriWhiteList_.contains(data.getScheme())`